controllers/cinder_controller.go

```diff
@@ -846,22 +846,12 @@ func (r *CinderReconciler) transportURLCreateOrUpdate(
 
 func (r *CinderReconciler) apiDeploymentCreateOrUpdate(ctx context.Context, instance *cinderv1beta1.Cinder) (*cinderv1beta1.CinderAPI, controllerutil.OperationResult, error) {
 
-	svcTemplate := instance.Spec.CinderAPI
 	cinderAPISpec := cinderv1beta1.CinderAPISpec{
-		ContainerImage:             svcTemplate.ContainerImage,
-		Replicas:                   svcTemplate.Replicas,
-		Debug:                      svcTemplate.Debug,
-		CustomServiceConfig:        svcTemplate.CustomServiceConfig,
-		DefaultConfigOverwrite:     svcTemplate.DefaultConfigOverwrite,
-		CustomServiceConfigSecrets: svcTemplate.CustomServiceConfigSecrets,
-		NetworkAttachments:         svcTemplate.NetworkAttachments,
-		ServiceUser:                instance.Spec.ServiceUser,
-		DatabaseHostname:           instance.Status.DatabaseHostname,
-		DatabaseUser:               instance.Spec.DatabaseUser,
-		Secret:                     instance.Spec.Secret,
-		PasswordSelectors:          instance.Spec.PasswordSelectors,
-		ExtraMounts:                instance.Spec.ExtraMounts,
-		TransportURLSecret:         instance.Status.TransportURLSecret,
+		CinderTemplate:     instance.Spec.CinderTemplate,
+		CinderAPITemplate:  instance.Spec.CinderAPI,
```

Comment:
+1 here, using `CinderAPITemplate` we can easily shorten the code here (same comment applies to the other services).